File: Producto/Backend/users-service/src/main/java/cl/duoc/pichangapp/users_service/dto/UserDTO.java
package cl.duoc.pichangapp.users_service.dto;

/**
 * DTO que representa la información pública de un usuario.
 * No expone el campo {@code id}: el identificador único visible para el cliente es el correo.
 * El id solo se usa internamente entre microservicios (vía JWT subject o path params).
 */
public record UserDTO(
        String correo,    // Correo electrónico (identificador único visible)
        String nombre,    // Nombre
        String apellido,  // Apellido
        boolean enabled,  // Estado de verificación de la cuenta
        String role       // Rol del usuario: "USER" o "ADMIN"
) {}

File: Producto/front-end/PichangApp/app/src/main/java/cl/duoc/pichangapp/data/model/UserDto.kt
package cl.duoc.pichangapp.data.model

data class UserDto(
    val id: String?,
    val nombre: String,
    val apellido: String,
    val correo: String? = null
)

data class UserUpdateRequest(
    val nombre: String,
    val apellido: String
)

data class PasswordUpdateRequest(
    val currentPassword: String,
    val newPassword: String
)
